### ISSUE TITLE
Router_par2   note velocity bug

### ISSUE THREAD

**Issue by BillS-B** [FluidSynth/fluidsynth#343]
Using the shell commands, 
router_begin note
router_par2 0 127 0 127
router_end
   should result in all notes playing at max volume -  so that a keyboard is no longer touch-sensitive.    This is needed for organ sounds.   But what happens is that the notes "attack" at full volume then immediately drop to the original par2 for the "sustain" and "decay"  - thus unusable for a church or other organs !   Try it with prog 0 20  !!!!
Hope this helps someone.  Running on a Raspberry Pi

**Comment by derselbst** [comment]
Which version of fluidsynth are you using?
What midi driver are you using?

> This is needed for organ sounds.

Why?

>  Try it with prog 0 20 !!!!

What soundfont ????

**Comment by BillS-B** [comment]
Thanks for your quick reply!

I am using the version that Raspberry Pi (B but not 3) has downloaded using 
apt-get install fluidsynth
It is version 1.1.5 apparently- is that the problem?  Yes I’m trying prog 0 20. Soundfont is the GM one loaded by Fluidsynth, for the moment [I will be using my own later]

Regarding the midi router – I don’t really know, I just used aconnect 20 128 to connect my keyboard to the fluidsynth client – is that wrong?  I got nowhere with jackctl.

Regards

Bill  PS I am an oldie, I wrote a realtime multitasking OS for the PDPs in 1971 that stayed in used  for years at Ford of Europe, but since then as an electronic etc etc tech director stuck with Pascal, Delphi, VB6 etc.  see www.chateau-rentals.com/Organ
I’ve been using the BASS libraries for my Windows organ sw, which is in VB6.

You help much appreciated as I want to scale down my sw for a headless Pi implementation.
I am new to Pi, & Linux.

**Comment by BillS-B** [comment]
started with
$ fluidsynth --audio-driver=alsa --gain 3 /usr/share/sounds/sf2/FluidR3_GM.sf2

**Comment by derselbst** [comment]
Well, we already have 1.1.9. Those are all bug fix releases. So before I
invest any time of looking into that, please get 1.1.9 and try again.

**Comment by BillS-B** [comment]
Thanks Tom,  but on my Raspberry 2B I can only install v1.1.5
I did sudo apt-get update
sudo apt-get install fluidsynth

Raspian/Debian may not have packaged the later versions?

Anyway I have tried again , adding router_clear  in front of the other commands.
This worked properly so I suppose there was a conflict with the “default” settings [which are not explained] ?

I’d avoided router_clear previously because it instantly rendered my keyboard dead - why?

Maybe the User Manual on GitHub could clarify this point !  It seems illogical that adding a Limit command can actually re-enable the midi input !!  Are the Limit commands actually essential routing commands??

Sadly I now find that fluidsynth crashes either itself or the whole OS if I start to play more than 3 or 4 notes at once.  My Rasperry has only 0.5Gb memory + 4Gb SD card.  I’m expecting to receive a new Pi 3B next week, which should add memory and processor power.  I’m living in hope. 
Thanks
Bill

**Comment by derselbst** [comment]
You always have to opportunity to install from another more up-to-date
repository or compile it yourself. Anyway, I wont invest my spare time by
giving support for a buggy, outdated software that was released 7 years ago.

Regarding memory: You can reduce mem usage by limiting fluidsynths
polyphony count, which defaults to 256, e.g.:

fluidsynth -o synth.polyphony=32

**Comment by derselbst** [comment]
While I was preparing 1.1.10, I looked into this. It's not a bug, it's a feature. When fluidsynth starts up it has a single midi router rule that simply passes through all events/notes as they get in. If you add another rule without calling `router_clear`, a single noteon will result in two noteons, which you can view by passing the `--dump` option via cmdline:

```
event_pre_noteon 0 62 1 // incoming event
event_post_noteon 0 62 127 // outgoing events
event_post_noteon 0 62 1
```

By the midi standard, the same key of the same channel cannot be "on" multiple times, thus the last noteon (with the original velocity) wins ultimately.

Long story short: just call `router_clear`. I will see if / where / how to update the documentation. Next time please consult the mailing list first. While there are only three active users at github, there are plenty of users on the list, probably many who are more familiar with the `midi_router` than myself.

**Comment by BillS-B** [comment]
Thanks Tom !
